src/webgpu/api/validation/encoding/cmds/render/draw.spec.ts

```diff
@@ -35,7 +147,53 @@ drawIndexedIndirect as it is GPU-validated.
     - range and GPUBuffer are both too small
 `
   )
-  .unimplemented();
+  .params(u =>
+    u
+      .combine('bufferSizeInElements', [10, 100])
+      // Binding size is always no larger than buffer size, make sure that setIndexBuffer succeed
+      .combine('bindingSizeInElements', [10])
+      .combine('drawIndexCount', [10, 11])
+      .combine('drawType', ['drawIndexed', 'drawIndexedIndirect'] as const)
+      .beginSubcases()
+      .combine('indexFormat', ['uint16', 'uint32'] as const)
+      .combine('useBundle', [false, true])
+  )
+  .fn(t => {
+    const {
+      indexFormat,
+      bindingSizeInElements,
+      bufferSizeInElements,
+      drawIndexCount,
+      drawType,
+      useBundle,
+    } = t.params;
+
+    const indexBuffer = t.createIndexBufferWithElementCount(bufferSizeInElements, indexFormat);
+    const indexElementSize = indexFormat === 'uint16' ? 2 : 4;
+    const bindingSize = bindingSizeInElements * indexElementSize;
+
+    const drawCall = new DrawCall(t);
+    drawCall.indexCount = drawIndexCount;
+    drawCall.drawType = drawType;
+
+    // Encoder finish will succeed if no index buffer access OOB when calling drawIndexed,
+    // and always succeed when calling drawIndexedIndirect.
+    const isFinishSuccess =
+      drawIndexCount <= bindingSizeInElements || drawType === 'drawIndexedIndirect';
+
+    const renderPipeline = t.createNaiveRenderPipeline();
+
+    const commandBufferMaker = t.createEncoder(useBundle ? 'render bundle' : 'render pass');
+    const renderEncoder = commandBufferMaker.encoder;
+
+    renderEncoder.setIndexBuffer(indexBuffer, indexFormat, 0, bindingSize);
+
+    renderEncoder.setPipeline(renderPipeline);
+
+    drawCall.callDraw(renderEncoder);
+
+    commandBufferMaker.validateFinish(isFinishSuccess);
```

Comment:
I think you should use validateFinishAndSubmit so we can also just be sure no validation fails in submit().

Reply:
Done. Sorry for forgetting the success case.